docker/Dockerfile

```diff
@@ -13,47 +13,140 @@
+    bison \
+    build-essential \
+    ca-certificates \
+    ccache \
+    clang-format-7 \
+    clang-tidy \
+    cron \
+    curl \
+    dnsutils \
+    gawk \
+    gdb \
+    git \
+    htop \
+    iproute2 \
+    lcov \
+    libcurl4-openssl-dev \
+    libsecp256k1-dev \
+    libssl-dev \
+    libtool \
+    libxml2-utils \
+    logrotate \
+    net-tools \
+    ninja-build \
+    nload \
+    ocl-icd-opencl-dev \
+    openssh-client \
+    pkg-config \
+    python3-dev \
+    python3-pip \
+    rsync \
+    rsyslog \
+    tar \
+    trickle \
+    unzip \
+    vim \
+    wget \
+    zip \
+    && apt-get remove -y cmake python2  && apt-get autoremove -y
+
+FROM deps AS builder
+
+# Cargo
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs -o install_script.sh && \
+    sh install_script.sh -y
+ENV PATH="/root/.cargo/bin:${PATH}"
+
+RUN cargo install sccache
+ENV SCCACHE_CACHE_SIZE="1G"
+ENV SCCACHE_DIR="/sccache"
+ENV RUSTC_WRAPPER="/root/.cargo/bin/sccache"
+
+# CMake 3.24
+ARG CMAKE_VERSION=3.24.2
+RUN wget https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-Linux-x86_64.sh \
+    && echo "739d372726cb23129d57a539ce1432453448816e345e1545f6127296926b6754 cmake-${CMAKE_VERSION}-Linux-x86_64.sh" | sha256sum -c \ 
+    && mkdir -p "${HOME}"/.local \
+    && bash ./cmake-${CMAKE_VERSION}-Linux-x86_64.sh --skip-license --prefix="${HOME}"/.local/ \
+    && "${HOME}"/.local/bin/cmake --version \
+    && rm cmake-${CMAKE_VERSION}-Linux-x86_64.sh
+ENV PATH="/root/.local/bin:${PATH}"
+
+# Setup ccache
+RUN ln -s "$(which ccache)" /usr/local/bin/gcc \
+  && ln -s "$(which ccache)" /usr/local/bin/g++ \
+  && ln -s "$(which ccache)" /usr/local/bin/cc \
+  && ln -s "$(which ccache)" /usr/local/bin/c++
+
+# This tag must be equivalent to the hash specified by "builtin-baseline" in vcpkg.json
+ARG VCPKG_COMMIT_OR_TAG=2022.09.27
+ARG VCPKG_ROOT=/vcpkg
 
-# run a copy -L to unfold the symlinkes, and strip all exes
-RUN mkdir -p /scilla/0/bin2/ && cp -L /scilla/0/bin/* /scilla/0/bin2/ && strip /scilla/0/bin2/*
+RUN ccache -p && ccache -z
 
-FROM zilliqa/zilliqa-ccache:${ZILLIQA_CCACHE_VERSION} AS builder
+# If COMMIT_OR_TAG is a branch name or a tag, clone a shallow copy which is
+# faster; if this fails, just clone the full repo and checkout the commit.
+RUN git clone https://github.com/microsoft/vcpkg ${VCPKG_ROOT} \
+  && git -C ${VCPKG_ROOT} checkout ${VCPKG_COMMIT_OR_TAG} \
+  && ${VCPKG_ROOT}/bootstrap-vcpkg.sh
 
-# Manually input tag or commit, can be overwritten by docker build-args
-ARG COMMIT_OR_TAG=v8.4.0
-ARG REPO=https://github.com/Zilliqa/Zilliqa.git
-ARG SOURCE_DIR=/root/zilliqa
-ARG BUILD_DIR=/root/zilliqa/build
-ARG INSTALL_DIR=/usr/local
-ARG VCPKG_ROOT=/vcpkg
+WORKDIR /zilliqa
+
+COPY vcpkg-registry/ vcpkg-registry
+COPY vcpkg-configuration.json .
+COPY vcpkg.json .
+
+RUN --mount=type=cache,target=/root/.cache/vcpkg/ ${VCPKG_ROOT}/vcpkg install --triplet=x64-linux-dynamic
```

Comment:
Comment about this?

Reply:
Forgot to ask previously: upon changing the dependencies which would trigger this; how does this work in parallel builds?

Reply:
It mounts the `/root/.cache/vcpkg` directory so it is shared between runs of `vcpkg install`. 

re parallel builds - https://docs.docker.com/engine/reference/builder/#run---mounttypecache
By default, it is used concurrently by multiple writers. I don't see a good reason to change this?

Reply:
If it works as expected and doesn't cause any issues I don't see a reason to change anything. I'm not familiar or sure how this works in the case of parallel builds and hope it wouldn't cause a recompilation of all the dependencies in each of those builds, hence the question.